app/controllers/qsets_controller.rb

```diff
@@ -3,17 +3,28 @@ class QsetsController < ApplicationController
 
   # shows all qsets
   def index
-    @qsets = @qsets.order(:name)
-    @question_counts = Question.all.group(:qset_id).count
+      if current_user.org.nil?
+        redirect_to root_url, alert: "You are not part of an organization."
+      else
+        redirect_to (current_user.org) 
+      end
   end
 
   # handles the request to show all questions in a qset
   def show
+    @question_counts = Question.all.group(:qset_id).count
     @feedback_active = !!current_user
     @questions = Question.includes(:answers).where(qset_id: @qset.id).order(created_at: :desc)
     @filter_mine = true if cookies[:all_mine_other_filter] == 'mine'
     @filter_other = true if cookies[:all_mine_other_filter] == 'other'
     @filter_all = true unless @filter_mine or @filter_other
+    if @qset.parent.nil?
```

Comment:
`@qsets` assignment is identical for the two code paths, it can be pulled out of the `if` block.